### ISSUE TITLE
powf overflow?

### ISSUE THREAD

**Issue by jacobrosenthal** [tarcieri/micromath#50]
-0.5881598.powf(2f32) over/underflowing to 489298620000.0 on thumbv7em-none-eabihf
On native its 0.3454914

Not sure where its going wrong. Luckily its far faster to compute with multiplication so I fixed mine, but seems like maybe some edge cases left to test.

Related, maybe pow(2) should short circuit to multiplication?

**Comment by tarcieri** [comment]
@Cazadorro any thoughts on this?

**Comment by Cazadorro** [comment]
I literally just never included the ability to take negative bases to a power... 🤦‍♂️ I think it was because I realized negative bases were weird and lots of other people never seemed to handle them correctly (windows calculator does not return the right value for, example. -0.5881598 ^ 3.2, it just copies the sign and does the calculation as if the base was positive, when it should be an imaginary number). 

Basically all exponents for negative bases are going to have to be integers otherwise this is going to return NaN. This is the same behavior that numpy and I think cmath employs (if you don't use complex type). 

Newest pull request should fix this issue.

**Comment by tarcieri** [comment]
Thanks @Cazadorro!

@jacobrosenthal can you confirm the issue is fixed on the `develop` branch? If so, I can cut a new release

**Comment by jacobrosenthal** [comment]
Yep!

**Comment by tarcieri** [comment]
Great! Released as v1.0.1
